File: app/src/main/java/cn/ntit/crps_app/model/StaticData.java
package cn.ntit.crps_app.model;

import com.google.gson.annotations.SerializedName;

public class StaticData {
    @SerializedName("mid") public String mid;
    @SerializedName("mm") public String mm;
    @SerializedName("msn") public String msn;
    @SerializedName("mloc") public String mloc;
    @SerializedName("mdate") public String mdate;
    @SerializedName("mrev") public String mrev;
    @SerializedName("prev") public int prev;
    @SerializedName("rimax") public float rimax;
    @SerializedName("rpmax") public float rpmax;
    @SerializedName("rt1") public float rt1;
    @SerializedName("rt2") public float rt2;

    public String getMidDisplay() {
        return cleanString(mid);
    }

    public String getMmDisplay() {
        return cleanString(mm);
    }

    public String getMsnDisplay() {
        return cleanString(msn);
    }

    public String getMlocDisplay() {
        return cleanString(mloc);
    }

    public String getMdateDisplay() {
        return cleanString(mdate);
    }

    public String getMrevDisplay() {
        return cleanString(mrev);
    }

    public String getPrevDisplay() {
        switch (prev) {
            case 0: return "V1.0";
            case 17: return "V1.1";
            case 34: return "V1.2";
            case 51: return "V1.3";
            case 255: return "未知";
            default: return String.format("(0x%02X)", prev);
        }
    }

    public String getRimaxDisplay() {
        return rimax == 0 ? "未知" : String.valueOf((int) rimax);
    }

    public String getRpmaxDisplay() {
        return rpmax == 0 ? "未知" : String.valueOf((int) rpmax);
    }

    public String getRt1Display() {
        return rt1 == 0 ? "未知" : String.valueOf((int) rt1);
    }

    public String getRt2Display() {
        return rt2 == 0 ? "未知" : String.valueOf((int) rt2);
    }

    /**
     * 去掉 # 和 0xFF 填充，空则返回"未知"
     */
    private String cleanString(String s) {
        if (s == null) return "未知";
        String cleaned = s.replace("#", "").replace("\u00FF", "").trim();
        return cleaned.isEmpty() ? "未知" : cleaned;
    }
}

File: app/src/main/java/cn/ntit/crps_compose/data/StaticData.kt
package cn.ntit.crps_compose.data

import org.json.JSONObject

/**
 * 静态电源信息：连接后周期性上报的 JSON。
 */
data class StaticData(
    val mid: String,
    val mm: String,
    val msn: String,
    val mloc: String,
    val mdate: String,
    val mrev: String,
    val prev: Int,
    val rimax: Float,
    val rpmax: Float,
    val rt1: Float,
    val rt2: Float,
) {
    val midDisplay: String get() = clean(mid)
    val mmDisplay: String get() = clean(mm)
    val msnDisplay: String get() = clean(msn)
    val mlocDisplay: String get() = clean(mloc)
    val mdateDisplay: String get() = clean(mdate)
    val mrevDisplay: String get() = clean(mrev)

    val prevDisplay: String get() = when (prev) {
        0 -> "V1.0"
        17 -> "V1.1"
        34 -> "V1.2"
        51 -> "V1.3"
        255 -> "未知"
        else -> String.format("(0x%02X)", prev)
    }

    /** 去掉 # 和 0xFF 填充，空则返回"未知" */
    private fun clean(s: String): String {
        val cleaned = s.replace("#", "").replace("\u00FF", "").trim()
        return cleaned.ifEmpty { "未知" }
    }

    companion object {
        fun parse(json: JSONObject): StaticData = StaticData(
            mid = json.optString("mid"),
            mm = json.optString("mm"),
            msn = json.optString("msn"),
            mloc = json.optString("mloc"),
            mdate = json.optString("mdate"),
            mrev = json.optString("mrev"),
            prev = json.optInt("prev"),
            rimax = json.optDouble("rimax").toFloat(),
            rpmax = json.optDouble("rpmax").toFloat(),
            rt1 = json.optDouble("rt1").toFloat(),
            rt2 = json.optDouble("rt2").toFloat(),
        )
    }
}
